core/group/service.go

```diff
@@ -99,13 +104,13 @@ func (s Service) AddUsers(ctx context.Context, groupIdOrSlug string, userIds []s
 		groupID = grp.ID
 	}
 
-	isAuthorized, err := s.relationService.CheckPermission(ctx, currentUser, namespace.DefinitionTeam, groupID, action.DefinitionManageTeam)
+	isAllowed, err := s.relationService.CheckPermission(ctx, currentUser, namespace.DefinitionTeam, groupID, action.DefinitionManageTeam)
 	if err != nil {
 		return []user.User{}, err
 	}
 
-	if !isAuthorized {
-		return []user.User{}, errors.Unauthorized
+	if !isAllowed {
```

Comment:
merge this if into the above error handling block ? (all error handling together feels better, for me at-least).

```golang
if err != nil {
   return nil, err
} else if !isAllowed {
   return nil, errors.ErrForbidden
}
```

```

Reply:
For this, even I think the `s.relationService.CheckPermission` only needs to return an error. Returning the status `isAlowed` or `isAuthorized` is not really needed. Inside, it could check end return error and we can check the type of error in this CheckPermission or maybe in the handler.
But again, don't want to change the logic that much.

I would actually prefer not to use `else` because sometimes it is confusing but I think we can merge the error handling block together.